scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -4384,16 +4384,16 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
     val mods = (implicitMod ++: annots(skipNewLines = true)) ++ localModifiers()
     if (mods forall {
         case _: Mod.Implicit | _: Mod.Lazy | _: Mod.Inline | _: Mod.Annot => true; case _ => false
-      })
-      (defOrDclOrSecondaryCtor(mods) match {
-        case stat if stat.isBlockStat => stat
-        case other => syntaxError("is not a valid block statement", at = other)
-      })
-    else
-      (tmplDef(mods) match {
-        case stat if stat.isBlockStat => stat
-        case other => syntaxError("is not a valid block statement", at = other)
-      })
+      }) {
```

Comment:
took me a minute to realize but very nice extraction of match to be used only once :+1: